### PR TITLE
fix: Txs Validate

### PR DESCRIPTION
**Pull request by lx-xiang**
The calculation method of tx size returned by calling proxyapp should be consistent with that of mempool

https://github.com/cometbft/cometbft/issues/1675
---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Comment by melekes** [comment]
Hey @lx-xiang! Thanks for looking into this :+1:. Did the fix solve your problem? From what I can see, with this fix we'll return an error in https://github.com/cometbft/cometbft/blob/ecb35fbb04efb0ecfa6491242e2af2d31192f802/state/execution.go#L155 during proposal block creation. Is that a desired outcome?

**Comment by lx-xiang** [comment]
@melekes The current fix does not solve my problem, but it will expose the issue in the Create ProposalBlock to avoid broadcasting invalid blocks，for cometbft, only this modification is needed

**Comment by melekes** [comment]
@lx-xiang could you please add a changelog entry in `.changelog`?

**Comment by lx-xiang** [comment]
I added

**Comment by lasarojc** [comment]
May have hit this bug in a test without malicious nodes.
The receiving side got too big a block:  `msg_type=*consensus.BlockPartMessage err="total size of proposal block parts exceeds maximum block bytes (4195546 > 4194304)"`

Confirming that the problem is not reproduced with the fix (applied to an internal branch).

**Review by lasarojc — APPROVED** [review]
The test missing in this PR is added in #1712

File: CHANGELOG.md [hunk 1]
```diff
@@ -47,6 +47,8 @@ gossip.
 
 - `[state/indexer]` Respect both height params while querying for events
    ([\#1529](https://github.com/cometbft/cometbft/pull/1529))
+- `[mempool]` The calculation method of tx size returned by calling proxyapp should be consistent with that of mempool
+  ([\#1687](https://github.com/cometbft/cometbft/pull/1687))
```
Comment (lasarojc) [inline CHANGELOG.md:51]:
```suggestion
```

**Comment by lasarojc** [comment]
unclog changelog also added in #1712

**Comment by lasarojc** [comment]
Thank you @lx-xiang . The fix is now merged into main and we'll back port it.

**Comment by lasarojc** [comment]
@Mergifyio refresh

**Pull request merged**